### DISCUSSION TITLE
ArgoCD application does not permit mutation by Kargo stage

### DISCUSSION THREAD

**Discussion by soudaburger** [akuity/kargo#2439] [Q&A] [answered]
Promotion Errored error executing Argo CD promotion mechanism: Argo CD Application "example" in namespace "argocd" does not permit mutation by Kargo Stage example in namespace example

We are getting the above error.  We believe it's an issue potentially with authentication between kargo and argocd, but not really sure.  It's also possible this is related to ArgoCD RBAC, but without knowing _how_ kargo is communicating with ArgoCD, it's a bit challenging to troubleshoot.  Any help here would be greatly appreciated.  Thank you.

**Comment by krancour** [comment, marked as answer]
@soudaburger my apologies... I thought this was documented better than it is, but in fact, I only find it in the quickstart's example, and it's not explained there:

Your Applications requires an annotation of the form `kargo.akuity.io/authorized-stage: {{project}}:{{stage}}`.

Kargo _itself_ has all the necessary permissions, but it looks for this annotation as a sign of consent from the Application to be modified _on behalf of_ a given Project + Stage.

I'll open an issue to close the documentation gap.

**Reply by soudaburger** [reply]
OMG, you're a life saver!!! Much appreciate the quick reply! Thanks!

**Reply by krancour** [reply]
I opened https://github.com/akuity/kargo/issues/2440 to track the required documentation improvement.
